### PR TITLE
Implement documents.create

### PR DESCRIPTION
**Pull request by zadjadr**
Implements https://www.getoutline.com/developers#tag/Documents/paths/~1documents.create/post and improves the pipeline

Related: https://github.com/ioki-mobility/go-outline/issues/5

### CODE REVIEW

File: .github/workflows/test.yml
Comment (StefMa) [inline .github/workflows/test.yml:24]: Why should we do this? Go test will download it automatically, no? 🤔

File: .github/workflows/test.yml
Comment (StefMa) [inline .github/workflows/test.yml:31]: We should also upload it somewhere right? Otherwise it doesn't make sense to generate it...

File: .github/workflows/test.yml
Comment (StefMa) [inline .github/workflows/test.yml:40]: What about to introduce this as a second workflow where we have such a filter in the 'on' section instead? 🤔

**Review by StefMa — COMMENTED** [review]
Didn't had a look at the code, but only about the CI improvement.

Did you saw #18?

File: .github/workflows/test.yml
Comment (zadjadr) [reply .github/workflows/test.yml:31]: Yes, I wasn't sure if we actually want to check coverage for now, but I'll add some uploading to it. Maybe github has some cool integrated feature where it can read theses coverage files and provide some graph/button. Saw a coverage button in other repos.

File: .github/workflows/test.yml
Comment (zadjadr) [reply .github/workflows/test.yml:40]: Makes sense, you mean a separate file, right?

File: .github/workflows/test.yml
Comment (zadjadr) [reply .github/workflows/test.yml:24]: I think this just makes our intentions and the stuff that happens "explicit". We can remove it if you like.

**Comment by zadjadr** [comment]
> Didn't had a look at the code, but only about the CI improvement.
> 
> Did you saw #18?

Now I saw, I'll wait until your PR is merged and then update this one (or separate the CI changes into another PR)

File: .github/workflows/test.yml
Comment (rsjethani) [reply .github/workflows/test.yml:24]: This is not necessary

File: .github/workflows/test.yml
Comment (rsjethani) [reply .github/workflows/test.yml:31]: Guys lets keep CI simple for now IMHO

File: .gitignore
Comment (rsjethani) [inline .gitignore:24]: No no no....nothing dev specific here...add it to your global `.gitconfig`

Comment (rsjethani) [reply .gitignore:24]: not everybody uses vscode ;)

**Comment by zadjadr** [comment]
I implemented documents.update as well & will create the PR, if this one is approved and merged

https://github.com/zadjadr/go-outline/pull/1

File: documents.go
Comment (StefMa) [inline documents.go:69]: Do we already want to go with such a complicated strucutre?
We need it earlier or later anyways, but so far we just went with the basics `create collection(name)`, that's it.
Just wanted to questions this, no changes requested.

Also, is it common to have so many parameters in go (in a single line)? 🤔 
If is so, I like the Kotlin code style to have each param in its own line:
```go
func (cl *DocumentsClient) Create(
    collectionID CollectionID, 
    title string,
    text string,
    ....,
```
But I have no strong opinion on that and don't know whats "the way" to go.
@rsjethani ? 🙂

File: documents.go
Comment (StefMa) [inline documents.go:78]: Do we really need this extensive documentation for private functions? 🤔 
Espeically because some of them are Self-explanation.
Like `collectionId` or `title` or `text` and so one 😁 

I would rather suggest to add the link to the outline API so in cae someone has to lookup something how the API works, we can press on it.

File: documents.go
Comment (StefMa) [inline documents.go:112]: Which specified `title` and `content`? 😉 
`Do` just makes the request for the given `DocumentsClientCreate` struct 😁

**Review by StefMa — COMMENTED** [review]
Code looks good so far 👌 Didn't tested it but should work 😉 
Just a few nitpitting questions/comments.

Thanks for the efford and the contribution 👍

File: documents.go
Comment (zadjadr) [reply documents.go:112]: Ups, probably copy paste :dancers: I'll also remove the extensive documentation :+1:

File: documents.go
Comment (zadjadr) [reply documents.go:67]: Do we already want to go with such a complicated strucutre?
- We (Sre) have a ticket, which needs us to be able to create a document with these features (only template + templateID where added as an extra, since it would be half implemented anyways)

Multiline parameters:
No Idea, I'd also be in favor of the kotlin way :)

Comment (rsjethani) [reply documents.go:67]: @zadjadr 
Multi-line parameters are usually not preferred, they are usually a code smell hinting that code restructuring can reduce number of parameters required. A very straight forward approach is to only ask for mandatory parameters. In this case you should take only `title` and `collection id` as compulsory values. For optional values you can have `option methods` which configure optional behavior. BTW you cannot have create method on DocumentsClient, you need a new type `DocumentsClientCreate` to maintain consistency across api usage. Following is how the API call can look like eventually:
```
client.Documents().Create(title, collection id).Text("doc text").Do(...)
```
Observe: `Text` is an example of a option method which allows configuring optional behavior. If we do not want to set text for the document then that call can be omitted.

**Review by StefMa — APPROVED** [review]
Yep, looks good to me 👍 
Nice work. 🚀

**Comment by StefMa** [comment]
@rsjethani please have a final look at this 🙏

File: models.go
Comment (StefMa) [inline models.go:7]: Not sure if if we need this as an extra type..
Maybe `DocumentId` is enough, but naming it `parentDocumentId` 🤷

File: models.go
Comment (StefMa) [inline models.go:32]: We could already create a `User` model 🤷

**Review by StefMa — APPROVED** [review]
Just two smaller comments.
However, I wil going to merge this as we should finally get things done.
Improvements can be still done later 👍 
I will create a follow-up issues for those things to be evaluated.

**Pull request merged**
